utils/swap_math.go

```diff
@@ -216,13 +273,13 @@ func calcIncrementalLiquidity(
 	} else {
 		// deltaL = (liquidity * currentSqrtP +/- absDelta) / nextSqrtP - liquidity
 		// needs to be minimum
-		tmp1 = MulDiv(liquidity, sqrtRatioCurrentX96, constants.Q96)
-		if exactIn {
-			tmp2 = new(big.Int).Add(tmp1, absAmount)
+		tmp1 = MulDiv(liquidity, currentSqrtP, constants.Q96)
+		if isExactInput {
+			tmp2 = new(big.Int).Add(tmp1, absDelta)
 		} else {
-			tmp2 = new(big.Int).Sub(tmp1, absAmount)
+			tmp2 = new(big.Int).Sub(tmp1, absDelta)
 		}
-		tmp3 = MulDiv(tmp2, constants.Q96, sqrtRatioTargetX96)
+		tmp3 = MulDiv(tmp2, constants.Q96, nextSqrtP)
```

Comment:
fix a bug here, correct code in contract is nextSqrtP, not targetSqrtP
https://github.com/KyberNetwork/ks-elastic-sc-v2/blob/1902450fd9bcbc39c8cf53b0570837513d32cdfb/contracts/libraries/SwapMath.sol#LL243C63-L243C63